### PR TITLE
update all dependencies

### PR DESCRIPTION
**Pull request by gambtho**
# Description

update all dependencies

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Is it a breaking change which will impact consuming tool(s)?

- [x] unit tests
- [x] e2e

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/controller/common/cleaner.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ func NewCleaner(manager ctrl.Manager, name controllername.ControllerNamer, gvrRe
 		return fmt.Errorf("creating clientset: %w", err)
 	}
 
-	mapper, err := apiutil.NewDynamicRESTMapper(manager.GetConfig(), apiutil.WithLazyDiscovery)
+	mapper, err := apiutil.NewDynamicRESTMapper(manager.GetConfig(), manager.GetHTTPClient())
```
Comment (OliverMKing) [inline pkg/controller/common/cleaner.go:48]: does the http client still use lazy discovery?

Worried about how many calls the rest mapper might make initially 

https://github.com/kubernetes-sigs/controller-runtime/pull/2116

File: pkg/controller/common/resource_reconciler_test.go [hunk 2]
```diff
@@ -150,6 +155,7 @@ func TestResourceReconciler_Start(t *testing.T) {
 		ObjectMeta: metav1.ObjectMeta{
 			Name:              "test",
 			DeletionTimestamp: &deletionTimeStamp,
+			Finalizers:        []string{"finalizer"},
```
Comment (OliverMKing) [inline pkg/controller/common/resource_reconciler_test.go:158]: i think we should just remove this test entirely. It's not useful and doesn't actually test anything with these additions

File: pkg/controller/common/cleaner.go (hunk 1 above)
Comment (gambtho) [reply pkg/controller/common/cleaner.go:48]: lazy by default apparently -- that entire block of code is removed
https://github.com/kubernetes-sigs/controller-runtime/blob/4332e3a54f000d642f3737f43d302ea6c5a91895/pkg/client/apiutil/restmapper_test.go#L84

File: pkg/controller/common/resource_reconciler_test.go (hunk 2 above)
Comment (gambtho) [reply pkg/controller/common/resource_reconciler_test.go]: dpne

**Review by OliverMKing — APPROVED** [review]

**Review by bfoley13 — APPROVED** [review]
looks good

**Pull request merged**